src/osgText/Text.cpp

```diff
@@ -49,6 +49,7 @@ Text::Text():
 {
     _supportsVertexBufferObjects = true;
 
+#ifndef OSG_GLES1_AVAILABLE
```

Comment:
I don't understand the need for this particular change. DisplaySettings::_textShaderTechnique is an empty string by default, so NO_TEXT_SHADER should be selected by default.  

I have strong preference for not putting platform specific hacks all over the OSG, we should only add them if we really really need them as every optional compile path is another build combination that needs to be maintained and could potentially break unseen for a long time.

Reply:
I agree adding the hack isn't a good solution, I'll take a closer look at why shaders are being added.  I haven't changed my client Text code since osg 3.2.1 so something with the Android build may have changed.

Reply:
Okay I see the problem: `NO_TEXT_SHADER` is never set on `_shaderTechnique` if the `_textShaderTechnique` is empty so it remains the default value, `GREYSCALE`. I'll explicitly set `NO_TEXT_SHADER` if the string is empty:

```
Text::Text():
    _shaderTechnique(GREYSCALE),
    _enableDepthWrites(true),
    _backdropType(NONE),
    _backdropHorizontalOffset(0.07f),
    _backdropVerticalOffset(0.07f),
    _backdropColor(0.0f, 0.0f, 0.0f, 1.0f),
    _colorGradientMode(SOLID),
    _colorGradientTopLeft(1.0f, 0.0f, 0.0f, 1.0f),
    _colorGradientBottomLeft(0.0f, 1.0f, 0.0f, 1.0f),
    _colorGradientBottomRight(0.0f, 0.0f, 1.0f, 1.0f),
    _colorGradientTopRight(1.0f, 1.0f, 1.0f, 1.0f)
{
    _supportsVertexBufferObjects = true;

    const std::string& str = osg::DisplaySettings::instance()->getTextShaderTechnique();
    if (!str.empty())
    {
        if (str=="ALL_FEATURES" || str=="ALL") _shaderTechnique = ALL_FEATURES;
        else if (str=="GREYSCALE") _shaderTechnique = GREYSCALE;
        else if (str=="SIGNED_DISTANCE_FIELD" || str=="SDF") _shaderTechnique = SIGNED_DISTANCE_FIELD;
        else if (str=="NO_TEXT_SHADER" || str=="NONE") _shaderTechnique = NO_TEXT_SHADER;
    }
    else
    {
        _shaderTechnique = NO_TEXT_SHADER;
    }

    assignStateSet();
}
```